python/pyarrow/io.pxi

```diff
@@ -1056,10 +1056,10 @@ cdef class Buffer(_Weakrefable):
             raise IndexError('Offset must be non-negative')
 
         if length is None:
-            result = SliceBuffer(self.buffer, offset)
+            result = GetResultValue(SliceBufferSafe(self.buffer, offset))
         else:
-            result = SliceBuffer(self.buffer, offset, max(length, 0))
-
+            result = GetResultValue(SliceBufferSafe(self.buffer, offset,
+                                                    max(length, 0)))
```

Comment:
Side question: should we rather raise an error for negative lengths? This `max(length, 0)` basically turns a negative length into an empty slice?

Reply:
Right, it would probably be a good idea to change this. I'll give it a try.